File: android/app/src/main/java/com/ivory/ivory/IvoryRecognitionService.java
package com.ivory.ivory;

import android.content.Intent;
import android.os.RemoteException;
import android.speech.RecognitionService;
import android.speech.SpeechRecognizer;

public class IvoryRecognitionService extends RecognitionService {
    @Override
    protected void onStartListening(Intent recognizerIntent, Callback callback) {
        try {
            callback.error(SpeechRecognizer.ERROR_NO_MATCH);
        } catch (RemoteException e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onCancel(Callback callback) {}

    @Override
    protected void onStopListening(Callback callback) {}
}

File: android/app/src/main/java/com/ivory/ivory/IvoryRecognitionService.kt
package com.ivory.ivory

import android.content.Intent
import android.os.RemoteException
import android.speech.RecognitionService
import android.speech.SpeechRecognizer

class IvoryRecognitionService : RecognitionService() {
    override fun onStartListening(recognizerIntent: Intent, callback: Callback) {
        try {
            // Immediately return no match
            callback.error(SpeechRecognizer.ERROR_NO_MATCH)
        } catch (e: RemoteException) {
            e.printStackTrace()
        }
    }

    override fun onCancel(callback: Callback) {}
    override fun onStopListening(callback: Callback) {}
}